### PR TITLE
Run3-sim116 Make use of printing and creating root tree compatible with multithreaded jobs in a number of packages of SimG4Core

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Make use of printing and creating root tree compatible with multithreaded jobs in a number of packages of SimG4Core

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
This is a step to modify geometry. We need this to stay in the main area and we have to build our geometry functions using that. Pruthvi will take care of other comments

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

File: SimG4Core/GFlash/TB/TreeMatrixCalib.cc [hunk 1]
```diff
@@ -1,87 +1,87 @@
...
+  myTree->Branch("tdcOffset", &myTdcOffset, "tdcOffset/D");
+  myTree->Branch("allMatrix", &myAllMatrix, "allMatrix/I");
+  myTree->Branch("amplit", &myAmplit, "amplit[49]/D");
+  myTree->Branch("crystal", &myCrystal, "crystal[49]/I");
 }
 
-TreeMatrixCalib::~TreeMatrixCalib() 
-{
+TreeMatrixCalib::~TreeMatrixCalib() {
+  /*
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeMatrixCalib.cc:33]: The commented out part can be removed

File: SimG4Core/GFlash/TB/TreeMatrixCalib.cc [hunk 2]
```diff
@@ -1,87 +1,87 @@
 #include "SimG4Core/GFlash/TB/TreeMatrixCalib.h"
 
-TreeMatrixCalib::TreeMatrixCalib(const char * filename) 
-{
-  myFile = new TFile(filename,"RECREATE");
-  myTree = new TTree("T1","my tree");
+TreeMatrixCalib::TreeMatrixCalib(const char* /*filename */) {
+  edm::Service<TFileService> fs;
+  //myFile = new TFile(filename,"RECREATE");
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeMatrixCalib.cc:5]: The commented out line can be removed, as well as the comment above in line 3

File: SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc [hunk 3]
```diff
@@ -1,215 +1,207 @@
...
-  Handle< EBRecHitCollection > pEBRecHits ;
-  const EBRecHitCollection*  EBRecHits = 0 ;
+  Handle<EBRecHitCollection> pEBRecHits;
+  const EBRecHitCollection* EBRecHits = 0;
   //try {
-  iEvent.getByLabel (RecHitProducer_, EBRecHitCollection_, pEBRecHits) ;
-  EBRecHits = pEBRecHits.product(); 
+  iEvent.getByLabel(RecHitProducer_, EBRecHitCollection_, pEBRecHits);
+  EBRecHits = pEBRecHits.product();
   //} catch ( std::exception& ex ) {
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc:96]: Is this commented out needed?

File: SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc [hunk 4]
```diff
@@ -1,215 +1,207 @@
...
 
   // taking what I need: hodoscopes
   Handle<EcalTBHodoscopeRecInfo> pHodo;
-  const EcalTBHodoscopeRecInfo* recHodo=0;
+  const EcalTBHodoscopeRecInfo* recHodo = 0;
   //try {
-  iEvent.getByLabel( hodoRecInfoProducer_, hodoRecInfoCollection_, pHodo);
+  iEvent.getByLabel(hodoRecInfoProducer_, hodoRecInfoCollection_, pHodo);
   recHodo = pHodo.product();
   //} catch ( std::exception& ex ) {
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc:107]: Is this commented out needed?

File: SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc [hunk 5]
```diff
@@ -1,215 +1,207 @@
...
   // taking what I need: tdc
   Handle<EcalTBTDCRecInfo> pTDC;
-  const EcalTBTDCRecInfo* recTDC=0;
+  const EcalTBTDCRecInfo* recTDC = 0;
   //try {
-  iEvent.getByLabel( tdcRecInfoProducer_, tdcRecInfoCollection_, pTDC);
-  recTDC = pTDC.product(); 
+  iEvent.getByLabel(tdcRecInfoProducer_, tdcRecInfoCollection_, pTDC);
+  recTDC = pTDC.product();
   //} catch ( std::exception& ex ) {
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc:118]: Is this commented out needed?

File: SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc [hunk 6]
```diff
@@ -1,215 +1,207 @@
...
   // taking what I need: event header
   Handle<EcalTBEventHeader> pEventHeader;
-  const EcalTBEventHeader* evtHeader=0;
+  const EcalTBEventHeader* evtHeader = 0;
   //try {
-  iEvent.getByLabel( eventHeaderProducer_ , pEventHeader );
-  evtHeader = pEventHeader.product(); 
+  iEvent.getByLabel(eventHeaderProducer_, pEventHeader);
+  evtHeader = pEventHeader.product();
   //} catch ( std::exception& ex ) {
```
Comment (perrotta) [inline SimG4Core/GFlash/TB/TreeProducerCalibSimul.cc:129]: Is this commented out needed?

**Review by perrotta — COMMENTED** [review]
Some further ckeaning could be allowed

**Comment by perrotta** [comment]
Ops, github did not show to me this PR as already merged...
Ok, I let the comments written anyhow, in case someone wants to apply some cleaning in these files. Otherwise, just ignore them.

**Pull request merged**